### DISCUSSION TITLE
Questions about the limits of the data I could send through BLE

### DISCUSSION THREAD

**Discussion by capc0307** [h2zero/NimBLE-Arduino#693] [Q&A] [answered]
How many characteristics could I include in a service? How many bytes (uint8_t) could I set in a characteristic (maximum length of the std::vector<uint8_t>)?

Thanks in advance,

capc0307.

**Comment by h2zero** [comment, marked as answer]
There is no hard limit on the number of characteristics. The BLE specification says the largest data size in a characteristic is 512 bytes.
